code/client/clrcore/External/Ped.cs

```diff
@@ -1599,5 +1599,109 @@ public static bool Exists(Ped ped)
 		{
 			return !ReferenceEquals(ped, null) && ped.Exists();
 		}
+
+		/// <summary>
+		/// Get the first entity in front of the relative Ped 
+		/// </summary>
+		/// <param name="distance">Max distance of the Raycast</param>
+		/// <returns>Returns the first Entity encountered in a distance specified</returns>
+		public Entity GetEntityInFront(float distance = 5f)
+		{
+			RaycastResult raycast = World.Raycast(Position, GetOffsetPosition(new Vector3(0f, distance, 0f)), IntersectOptions.Everything);
+			if (raycast.DitHitEntity)
+				return (Entity)raycast.HitEntity;
+			return null;
+		}
+
+		/// <summary>
+		/// Get closest Peds to the specified Ped
+		/// </summary>
+		/// <returns></returns>
+		public Tuple<float, Ped> GetClosestPed()
+		{
+			Ped[] Peds = World.GetAllPeds();
+			float closestDistance = -1f;
+			Ped closestPed = null;
+			foreach (Ped v in Peds)
+			{
+				Vector3 pedcoords = v.Position;
+				float distance = World.GetDistance(pedcoords, Position);
+				if (closestDistance == -1 || closestDistance > distance)
+				{
+					closestPed = v;
+					closestDistance = distance;
+				}
+			}
+			return new Tuple<Ped, float>(closestDistance, closestPed);
+		}
+		
+		/// <summary>
+		/// Get the closest Ped to the specified coords
+		/// </summary>
+		/// <param name="Coords"></param>
+		/// <returns></returns>
+		public Tuple<float, Ped> GetClosestPed(Vector3 Coords)
+		{
+			Ped[] Peds = World.GetAllPeds();
+			float closestDistance = -1f;
+			Ped closestPed = null;
+			foreach (Ped v in Peds)
+			{
+				Vector3 pedcoords = v.Position;
+				float distance = World.GetDistance(pedcoords, Coords);
+				if (closestDistance == -1 || closestDistance > distance)
+				{
+					closestPed = v;
+					closestDistance = distance;
+				}
+			}
+			return new Tuple<float, Ped>(closestDistance, closestPed);
+		}
+
+		/// <summary>
+		/// Get closest vehicle at the specified coords
+		/// </summary>
+		/// <param name="Coords"></param>
+		/// <returns></returns>
+		public Tuple<float, Vehicle> GetClosestVehicle(Vector3 Coords)
+		{
+			Vehicle[] vehs = World.GetAllVehicles();
+			float closestDistance = -1f;
+			Vehicle closestVehicle = null;
+			foreach (Vehicle v in vehs)
+			{
+				Vector3 vehcoords = v.Position;
+				float distance = World.GetDistance(vehcoords, Coords);
+				if (closestDistance == -1 || closestDistance > distance)
+				{
+					closestVehicle = v;
+					closestDistance = distance;
+				}
+			}
+			return new Tuple<float, Vehicle>(closestDistance, closestVehicle);
+		}

```

Comment:
```cs
public Tuple<float, Vehicle> GetClosestVehicle(Vector3 Coords)
{
    Vehicle[] vehs = World.GetAllVehicles();
    float closestDistance = Single.MaxValue;
    
    Vehicle closestVehicle = null;
    foreach (Vehicle v in vehs)
    {
        float distanceSquared = Vector3.DistanceSquared(v.Position, Coords);
        
        if (distanceSquared < closestDistance)
        {
            closestVehicle = v;
            closestDistance = distanceSquared;
        }
    }
    return new Tuple<float, Vehicle>((float)Math.Sqrt(closestDistance), closestVehicle);
}
```
Avoid equality comparisons on floats like that, anyway you can compare distance using squared ones which should be faster.

Reply:
Once added this, you can avoid writing duplicated code for the version which uses the Ped.Position property instead and just use
```cs
public Tuple<float, Vehicle> GetClosestVehicle()
{
    return GetClosestVehicle(Position);
}
```

Reply:
And then do the same for the `GetClosestPed` methods.

Reply:
i see 🤔 i'll do it now

Reply:
If you check the [`World`](https://github.com/citizenfx/fivem/blob/master/code/client/clrcore/External/World.cs) class, you'll notice similar API are already there, maybe you could add a method to the Entity class instead which will just invoke those ones.

Reply:
yeah that's actually true.. i didn't see it because it was commented out.. i'll edit that!!